src/main/java/org/broadinstitute/hellbender/tools/walkers/ReferenceConfidenceVariantContextMerger.java

```diff
@@ -194,18 +201,18 @@ private VCWithNewAlleles(final VariantContext vc, final List<Allele> newAlleles,
             this.isSpanningEvent = isSpanningEvent;
         }
 
-        public List<Allele> filterAllelesForFinalSet() {
+        public Stream<Allele> filterAllelesForFinalSet() {
             return newAlleles.stream().filter(a -> !a.equals(GATKVCFConstants.NON_REF_SYMBOLIC_ALLELE))
                     .filter(a -> !a.isReference())
                     .filter(a -> !(a.isSymbolic() && vc.isSymbolic())) // skip <*DEL> if there isn't a real alternate allele.
-                    .filter(Allele::isCalled) // skip NO_CALL
-                    .collect(Collectors.toList());
+                    .filter(Allele::isCalled) ; // skip NO_CALL
         }
 
         // record whether it's also a spanning deletion/event (we know this because the VariantContext type is no
         // longer "symbolic" but "mixed" because there are real alleles mixed in with the symbolic non-ref allele)
         public boolean isSpanningDeletion() {
```

Comment:
I've added an exhaustive test. This method is now better tested than the majority of the genotyping engine.